### PR TITLE
Make test data file paths relative to project root.

### PR DESCRIPTION
**Pull request by jonathanknowles**
# Issue Number

None.

# Overview

This change adjusts our test suites so that paths of test data files (such as `block0.bin`) are relative to the _project root_, rather than relative to the _current working directory_.

This makes it possible to run unit and integration tests from `stack ghci` and `ghcid`.

## Example

```
ghcid --command 'stack ghci cardano-wallet-jormungandr:test:integration' --test ':main -m decode'
```
Without this PR, we see errors similar to:
```
*** Exception: test/data/jormungandr/block0.bin: openBinaryFile: does not exist (No such file or directory)
```

### CODE REVIEW

**Comment by KtorZ** [comment]
_didn't we agree to do this during a recovery week :grimacing:_ ?

**Comment by rvl** [comment]
I will review it in recovery week. :-)

I do appreciate being able to run tests under ghci. Previously I had to add a symlink in my local tree.

**Review by KtorZ — APPROVED** [review]
Reworked a bit, seems to work fine for me with ghcid now.

File: lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/Launch.hs [hunk 1]
```diff
@@ -44,16 +43,19 @@ withConfig = bracket setupConfig teardownConfig
 
 setupConfig :: IO JormungandrConfig
 setupConfig = do
+    block0 <- getDataFileName "jormungandr/block0.bin"
+    secret <- getDataFileName "jormungandr/secret.yaml"
+    config <- getDataFileName "jormungandr/config.yaml"
```
Comment (jonathanknowles) [inline lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/Launch.hs:48]: Shouldn't we be using `</>` here?

Comment (KtorZ) [reply lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/Launch.hs:48]: Apparently no. Unless maybe we are compiling from windows. It's quite unclear from the cabal doc but it seems that in code, this really acts as a label and `getDataFileName` resolves it to a name that is compatible with the underlying operating system:

(source: https://www.haskell.org/cabal/users-guide/developing-packages.html?highlight=data%20dir#accessing-data-files-from-package-code)

> If the argument is a filename listed in the data-files field, the result is the name of the corresponding file on the system on which the program is running.

**Pull request merged**